app/rtc_s1_parallel.py

```diff
@@ -0,0 +1,203 @@
+import os
+import subprocess
+import time
+import yaml
+
+from rtc.runconfig import RunConfig
+
+
+def get_rtc_s1_parser():
+    '''Initialize YamlArgparse class and parse CLI arguments for OPERA RTC.
+    Modified after copied from `rtc_s1.py`
+    '''
+    parser = argparse.ArgumentParser(description='',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('run_config_path',
+                        type=str,
+                        default=None,
+                        help='Path to run config file')
+
+    # Determine the default # of concurrent workers    
+    ncpu_default = os.cpu_count()
+    if os.getenv('OMP_NUM_THREADS') is not None:
+        omp_num_threads = int(os.getenv('OMP_NUM_THREADS'))
+        ncpu_default = min(ncpu_default, omp_num_threads)
+
+    parser.add_argument('-n',
+                        dest='num_workers',
+                        type=int,
+                        default=ncpu_default,
+                        help='Number of concurrent workers.')
+
+    parser.add_argument('--log',
+                        '--log-file',
+                        dest='log_file',
+                        type=str,
+                        help='Log file')
+
+    parser.add_argument('--full-log-format',
+                        dest='full_log_formatting',
+                        action='store_true',
+                        default=False,
+                        help='Enable full formatting of log messages')
+
+    return parser
+
+def split_runconfig(path_runconfig_in, path_log_in):
+    '''
+    Split the input runconfig into single burst runconfigs.
+    Writes out the runconfigs.
+    Return the list of the burst runconfigs.
+
+    Parameters:
+    path_runconfig_in: str
+        Path to the original runconfig
+    path_log_in: str
+        Path to the original logfile
+
+    Returns:
+    list_runconfig_burst: list(str)
+        List of the burst runconfigs
+    list_logfile_burst: list(str)
+        List of the burst logfiles,
+        which corresponds to `list_runconfig_burst`
+    '''
+    with open(path_runconfig_in, 'r+', encoding='utf8') as fin:
+        runconfig_dict_in = yaml.safe_load(fin.read())
+
+    cfg = RunConfig.load_from_yaml(path_runconfig_in, 'rtc_s1')
+
+
+    list_runconfig_burst = []
+    list_logfile_burst = []
+
+    # determine the bursrs to process
+    list_burst_id = cfg.bursts.keys()
+
+    for burst_id in list_burst_id:
+        path_temp_runconfig = os.path.join(cfg.scratch_path,
+                                           f'burst_runconfig_{burst_id}.yaml')
+
+        runconfig_dict_out = runconfig_dict_in.copy()
+        set_dict_item_recursive(runconfig_dict_out,
+                                ['runconfig',
+                                 'groups',
+                                 'input_file_group',
+                                 'burst_id'],
+                                [burst_id])
+
+        set_dict_item_recursive(runconfig_dict_out,
```

Comment:
This script loads the user-provided runconfig as it is, and split them into single bursts.
I think some users would not set up some values in the input runconfig (e.g. `memory_mode`). In that case, accessing the nonexistent item would cause an error. To deal with such case, I've written `set_dict_item_recursive()`.